File: C#/19_RemoveNthNodeFromEndOfList.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        
        List<ListNode> lst = new List<ListNode>();
        ListNode it = new ListNode();
        it.next = head.next;
        
        lst.Add(head);
        while(it.next != null){
            lst.Add(it.next);
            it = it.next;
        }
        
        int k = lst.Count - n -1;
        
        if (k < 0) 
            return lst[0].next;
        
        lst[k].next = lst[k].next.next;
        return head;
    }
}

File: Python/19_RemoveNthNodeFromEndOfList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        it, count = head, 1
        p0 = ListNode()
        p0.next = head
        while it.next is not None:
            if count == n:
                p0 = p0.next
            else:
                count += 1
                        
            it = it.next
        
        if p0.next == head:
            return head.next                    
        else:
            p0.next = p0.next.next
            
        return head
                